frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/SlideInBackgroundable.java

```diff
@@ -0,0 +1,166 @@
+  private final IBranchLayout branchLayout;
+  private final IBranchLayoutWriter branchLayoutWriter;
+  private final Runnable preSlideInRunnable;
+  private final SlideInOptions slideInOptions;
+  private final String parentName;
+  private final VcsNotifier notifier;
+
+  public SlideInBackgroundable(
+      Project project,
+      GitRepository gitRepository,
+      IBranchLayout branchLayout,
+      IBranchLayoutWriter branchLayoutWriter,
+      Runnable preSlideInRunnable,
+      SlideInOptions slideInOptions,
+      String parentName) {
+    super(project, getString("action.GitMachete.BaseSlideInBranchBelowAction.task-title"));
+    this.preSlideInRunnable = preSlideInRunnable;
+    this.gitRepository = gitRepository;
+    this.branchLayout = branchLayout;
+    this.slideInOptions = slideInOptions;
+    this.parentName = parentName;
+    this.notifier = VcsNotifier.getInstance(project);
+    this.branchLayoutWriter = branchLayoutWriter;
+  }
+
+  @Override
+  public void run(ProgressIndicator indicator) {
+    preSlideInRunnable.run();
+
+    waitForLocalBranch();
+
+    Path macheteFilePath = getMacheteFilePath(gitRepository);
+
+    var childEntryByName = branchLayout.findEntryByName(slideInOptions.getName());
+    IBranchLayoutEntry entryToSlideIn;
+    IBranchLayout targetBranchLayout;
+    if (childEntryByName.isDefined()) {
+
+      if (slideInOptions.shouldReattach()) {
+        entryToSlideIn = childEntryByName.get();
+        targetBranchLayout = branchLayout;
+      } else {
+        entryToSlideIn = childEntryByName.map(e -> e.withChildren(List.empty())).getOrNull();
+        targetBranchLayout = Try.of(() -> branchLayout.slideOut(slideInOptions.getName()))
+            .onFailure(e -> Match(e).of(
+                Case($(instanceOf(EntryDoesNotExistException.class)), exceptionWithMessageHandler(
+                    format(
+                        getString(
+                            "action.GitMachete.BaseSlideInBranchBelowAction.notification.message.entry-does-not-exist"),
+                        entryToSlideIn.getName()))),
+                Case($(instanceOf(EntryIsRootException.class)), exceptionWithMessageHandler(
+                    format(getString("action.GitMachete.BaseSlideInBranchBelowAction.notification.message.entry-is-root"),
+                        entryToSlideIn.getName()))),
+                Case($(), exceptionWithMessageHandler(/* message */ null))))
+            .getOrNull();
+
+        if (targetBranchLayout == null) {
+          return;
+        }
+      }
+
+    } else {
+      entryToSlideIn = new BranchLayoutEntry(slideInOptions.getName(), /* customAnnotation */ null,
+          /* children */ List.empty());
+      targetBranchLayout = branchLayout;
+    }
+
+    var newBranchLayout = Try
+        .of(() -> targetBranchLayout.slideIn(parentName, entryToSlideIn))
+        .onFailure(e -> Match(e).of(
+            Case($(instanceOf(EntryDoesNotExistException.class)), exceptionWithMessageHandler(
```

Comment:
Maybe use `onFailure(Class<X> exceptionType, Consumer<? super X> action)` instead to simply things a bit